### DISCUSSION TITLE
Test changes to renovate configuration?

### DISCUSSION THREAD

**Discussion by janpio** [renovatebot/renovate#25604] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

When making changes to the renovate configuration in our repositories, we usually do a PR to the `renovate.json`, merge it, and then wait for Renovate to do its thing - sometimes sped up via "Rebase all open PRs" or similar. 

That of course is suboptimal as any mistake or omission requires another PR, and the same dance again.

Is there a way to test the new Renovate configuration before merging the change? 

I could imagine that the PR that changes the configuration can request a debug run of some kind that then shows what PRs with which changes would be created after the configuration change is merged.

### What you tried so far.

-

### Relevant debug logs

-

**Comment by rarkins** [comment]
Please try this: https://docs.renovatebot.com/getting-started/installing-onboarding/#reconfigure-via-pr

**Reply by HonkingGoose** [reply]
The docs are not updating properly right now, but Renovate `37.44.0` added support for this:

> ### Validation of Renovate config change PRs
> 
> Renovate can validate configuration changes in Pull Requests when you use a special branch name.
> 
> Follow these steps to validate your configuration:
> 
> 1. Create a new Git branch that matches the `{{branchPrefix}}reconfigure` pattern. For example, if you're using the default prefix `renovate/`, your branch name must be `renovate/reconfigure`.
> 1. Commit your updated Renovate config file to this branch, and push it to your Git hosting platform.
> 
> The next time Renovate runs on that repo it will:
> 
> 1. Search for a branch that matches the special reconfigure pattern.
> 1. Check for a config file in the reconfigure branch. Renovate can even find a renamed configuration file (compared to the config file in the default branch).
> 1. Add a passing or failing status to the branch, depending on the outcome of the config validation run.
> 1. If there's an _open_ pull request with validation errors from the _reconfigure_ branch then Renovate comments in the PR with details.
> 1. Validate each commit the next time Renovate runs on the repository, until the PR is merged.

Links:

- https://github.com/renovatebot/renovate/pull/24699
- https://github.com/renovatebot/renovate/blob/main/docs/usage/config-validation.md#validation-of-renovate-config-change-prs

**Reply by janpio** [reply]
Thanks, that is already pretty cool and the mechanism to run on special branch names will work great for us.

I did understand correctly that this will "only" validate, and not tell me the resulting changes this will trigger after merge, correct?

**Reply by RahulGautamSingh** [reply, marked as answer]
Yes, the latter will be added as a part of: https://github.com/renovatebot/renovate/issues/547
